src/libcryptobox/cryptobox.c

```diff
@@ -630,7 +645,7 @@ void rspamd_cryptobox_sign(unsigned char *sig, unsigned long long *siglen_p,
 		g_assert(EVP_DigestSignInit(sha_ctx, &pctx, EVP_sha512(),
 									ENGINE_F_ENGINE_NEW, pkey) == 1);
 
-		g_assert(EVP_DigestSignFinal(sha_ctx, sig, &diglen) == 1);
+		g_assert(EVP_DigestSignFinal(sha_ctx, sig, (size_t *) &diglen) == 1);
```

Comment:
It's not safe to cast pointers to integers of different size. I would say, that either `diglen` should be changed to `size_t` or there should be a dedicated local variable of `size_t` type for that purposes (if the first option is not possible only).